core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+
+        Args:
+            blog_post_id: str. The blog post ID to validate.
+        """
+        if not isinstance(blog_post_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Blog Post ID should be a string, received: %s' % blog_post_id)
+
+        if len(blog_post_id) != constants.BASE_MODELS_ID_LENGTH:
+            raise utils.ValidationError('Invalid Blog Post ID.')
+
+    @classmethod
+    def require_valid_url_fragment(cls, url_fragment):
+        """Checks whether the url fragment of the blog post is a valid one.
+
+        Args:
+            url_fragment: str. The url fragment to validate.
+        """
+
+        url_limit = constants.MAX_CHARS_IN_BLOG_POST_URL_FRAGMENT
+        utils.require_valid_url_fragment(
+            url_fragment, 'Blog Post URL Fragment', url_limit)
+
+    @classmethod
+    def require_valid_title(cls, title, strict):
+        """Checks whether the blog post title is a valid one.
+
+        Args:
+            title: str. The title to validate.
+            strict: bool. Enable strict checks on the blog post summary when the
+                blog post is published or is going to be published.
+        """
+
+        if not isinstance(title, python_utils.BASESTRING):
+            raise utils.ValidationError('Title should be a string.')
+
+        title_limit = constants.MAX_CHARS_IN_BLOG_POST_TITLE
+        if len(title) > title_limit:
+            raise utils.ValidationError(
+                'blog post title should at most have %d chars, received: %s'
+                % (title_limit, title))
+
+        if strict:
+            if len(title) == 0:
+                raise utils.ValidationError('Title should not be empty')
+
+    @classmethod
+    def require_valid_tags(cls, tags, strict):
+        """Validates tags for the blog post object.
+
+        Args:
+            tags: list(str). The list of tags assigned to a blog post.
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+
+        Raises:
+            ValidationErrors.
+        """
+        if not isinstance(tags, list):
+            raise utils.ValidationError(
+                'Expected \'tags\' to be a list, received: %s' % tags)
+        for tag in tags:
+            if not isinstance(tag, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected each tag in \'tags\' to be a string, received: '
+                    '\'%s\'' % tag)
+
+            if len(tag) == 0:
+                raise utils.ValidationError(
+                    'Tag should not be empty, received: \'\'')
+
+            if not re.match(constants.TAG_REGEX, tag):
+                raise utils.ValidationError(
+                    'Tags should only contain lowercase letters and spaces, '
+                    'received: \'%s\'' % tag)
+
+            if (tag[0] not in string.ascii_lowercase or
```

Comment:
Ditto, please use a regex to check for this.

Reply:
Done.